src/util/HttpServer/ContentEncodingHelper.h

```diff
@@ -0,0 +1,46 @@
+// Copyright 2022, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Robin Textor-Falconi (textorr@informatik.uni-freiburg.de)
+
+#pragma once
+
+#include "./beast.h"
+
+namespace ad_utility::content_encoding {
+
+enum class CompressionMethod { NONE, DEFLATE, GZIP };
+
+namespace detail {
+
+constexpr CompressionMethod getCompressionMethodAcceptEncodingHeader(
+    const std::string_view acceptEncodingHeader) {
+  // TODO evaluate if gzip can be supported using boost
+  if (acceptEncodingHeader.find("deflate") != std::string::npos) {
+    return CompressionMethod::DEFLATE;
+  } else if (acceptEncodingHeader.find("gzip") != std::string::npos) {
+    return CompressionMethod::GZIP;
+  }
+  return CompressionMethod::NONE;
+}
+}  // namespace detail
+
+template <typename Body>
+inline CompressionMethod getCompressionMethodForRequest(

```

Comment:
`inline` is redundant for `template`d functions, is this an artifact from experiments, or is there a certain rationale?

Reply:
This is an artifact from an earlier version of this function